feat: new configure project page for Maven (#5455)
JetBrains/kotlin-web-site: docs/topics/maven/maven-configure-project.md

[//]: # (title: Configure a Maven project)

When you introduce Kotlin to your existing Java Maven project or create a new Kotlin Maven project, you need
to add the Kotlin Maven plugin that compiles Kotlin sources and modules.

Currently, only Maven v3 is supported.

## Automatic configuration with extensions

You can simplify Maven configuration in both mixed Java-Kotlin projects and in pure Kotlin projects using the `<extensions>` option.

> It allows skipping the Maven compiler plugin configuration.

Reviewer: ```suggestion
This approach saves you time because you don't need to configure the Maven compiler plugin.
```
I think this version makes it clearer what the benefit is. WDYT?

   </properties>
   ```

2. In the `<build><plugins>` section, add the Kotlin Maven plugin with the enabled `<extensions>` option:

   ```xml
   <build>
       <plugins>
           <!-- Kotlin compiler plugin configuration -->
           <plugin>
               <groupId>org.jetbrains.kotlin</groupId>
               <artifactId>kotlin-maven-plugin</artifactId>
               <version>${kotlin.version}</version>
               <extensions>true</extensions> <!-- Enable the extension -->
           </plugin>
           <!-- No need to configure Maven compiler plugin with extensions -->
       </plugins>
   </build>
   ```

The `<extensions>` option:

* Registers `src/main/kotlin` and `src/test/kotlin` directories as source roots if they already exist but are not specified in the plugin configuration.
* Adds the [`kotlin-stdlib` dependency](maven-set-dependencies.md#dependency-on-the-standard-library) if it's not already defined in the project.
* Adds `compile`, `test-compile`, `kapt`, and `test-kapt` executions to your build, bound to their appropriate [lifecycle phases](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html).

>   So, you don't need to manually set up the `<executions>` section with `<id>` and `<goals>`.

Reviewer: ```suggestion
  So you don't need to manually set up the `<executions>` section with `<id>` and `<goals>`.
```

> the `<build>` section has priority for lifecycle settings. All earlier changes to lifecycle settings are ignored.
>
{style="note"}

### Change Maven compiler version

Currently, the default version of the Maven compiler plugin used with `<extensions>` is **%mavenExtensionsVersion%**.
You can set a different version separately:

```xml
<build>
    <plugins>
        <!-- Kotlin compiler plugin configuration -->
        <plugin>
            <groupId>org.jetbrains.kotlin</groupId>
            <artifactId>kotlin-maven-plugin</artifactId>
            <version>${kotlin.version}</version>
            <extensions>true</extensions>
        </plugin>
        <!-- Maven compiler plugin configuration for Java classes -->
        <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-compiler-plugin</artifactId>
            <version>%mavenPluginVersion%</version>
        </plugin>
    </plugins>
</build>
```

## Manual configuration without extensions

Without enabling `<extensions>` in the Kotlin Maven plugin, a manual project configuration is necessary to ensure that

> the source code is compiled correctly.

Reviewer: ```suggestion
Without enabling `<extensions>` in the Kotlin Maven plugin, you need to configure the project manually to ensure that
the source code compiles correctly.
```
